### ISSUE TITLE
DIFF CHD ERROR: file not found

### ISSUE THREAD

**Issue by tulu** [libretro/mame#37]
When trying to load a game with a CHD file (placed correctly) in the logs I get the error: 

```
kinst.chd DIFF CHD ERROR: file not found
FATALERROR: Required files are missing, the machine cannot be run.
```
My folder structure is:
/roms/mame/kinst.zip
/roms/mame/kinst/kinst.chd

The above folder structure works fine for Mameui64.
I've read about a diff folder and trying to delete what is generated by mame but I couldn't find it.

**Comment by fr500** [comment]
I just tried in the latest core and it works for me

**Comment by hizzlekizzle** [comment]
yeah, works for me with ../romdir/kinst.zip and ../romdir/kinst/kinst.chd with chd md5sum of:
8477ed69164c81edf2842c38c2c868dc

**Comment by tulu** [comment]
I have a chd file matching the md5sum. 
I'm running RetroArch 1.2.2 win-x64 with the mame_libretro core updated 3 days ago.
I'm guessing that it can't find the file with the current folder structure.

**Comment by r-type** [comment]
late but ...
the diff is create  generally in retroarch_sysdir/mame/diff.
then it will create the diff chd in this folder.

**Comment by tulu** [comment]
I've just downloaded RetroArch-v1.3.0-x86_64-Windows and the game worked fine. Since October I didn't try to play it again. So either it was a problem with the version I was using, the core I was using or the installation I made at that time.

**Comment by tulu** [comment]
OK, so I´ve found the issue. If I set system_directory = "default" in retroarch.cfg Killer Instinct starts to work. The thing is that my bios are in :\bios, so when I change this in the settings, Reicast and PPSSPP start working but Killer Instinct won´t... Am I missing something?

**Comment by fr500** [comment]
I guess mame is not being able to create the directories in bios, try creating a mame folder in bios and set it back to that, create bios\mame\ini and bios\mame\hash at least

**Comment by tulu** [comment]
I´ve tried that but I got the same error. If I set default to system_directory, where I should put the bios files?

**Comment by tulu** [comment]
OK, so setting the system_directory to :\system and adding there the new folders /mame/diff made both Killer Instinct and Reicast work (with their bios in the system dir). Is still weird that when you change the default settings some things works and others doesn't... If I set default for system_directory the bios files can't be read...

**Comment by fr500** [comment]
it's not the setting, it's just the core being dumb and not being able to create the dirs.
It works for me on <content dir>

**Comment by fr500** [comment]
I just checked the functionality with <content dir> and with a manually set dir on both mame, and reicast and it's working fine granted you have the BIOS files there.

With <content dir>:
For reicast I copied my DC folder in myrompath\dreamcast and loaded and it booted fine.
For mame I copied my mame folder to myrompath\arcade\mame and it booted fine, I also confirmed by checking the log


RetroArch [libretro DEBUG] ::  -cfg_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\cfg
RetroArch [libretro DEBUG] ::  -nvram_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\nvram
RetroArch [libretro DEBUG] ::  -hiscore_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\hi
RetroArch [libretro DEBUG] ::  -input_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\input
RetroArch [libretro DEBUG] ::  -state_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\states
RetroArch [libretro DEBUG] ::  -snapshot_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\snaps
RetroArch [libretro DEBUG] ::  -diff_directory
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Saves\MAME\mame\diff
RetroArch [libretro DEBUG] ::  -samplepath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014\\mame\samples
RetroArch [libretro DEBUG] ::  -artpath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014\\mame\artwork
RetroArch [libretro DEBUG] ::  -cheatpath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014\\mame\cheat
RetroArch [libretro DEBUG] ::  -inipath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014\\mame\ini
RetroArch [libretro DEBUG] ::  -hashpath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014\\mame\hash
RetroArch [libretro DEBUG] ::  -rompath
RetroArch [libretro DEBUG] ::  ..\..\EmulatorData\Games\Multi\MAME 2014

And it set the paths automatically as intended.
With system dir set to a discrete it worked too of course and no further debugging was needed.
